paramiko/sftp_client.py

```diff
@@ -780,10 +780,10 @@ def getfo(
             the bytes transferred so far and the total bytes to be transferred
         :param bool prefetch:
             controls whether prefetching is performed (default: True)
-        :param int max_outstanding_prefetch_requests:
-            Limit the number of outstanding requests instead of sending all
-            at once, to avoid problems with servers that drop requests when
-            their queue is full.
+        :param int max_concurrent_prefetch_requests:
+            The maximum number of concurrent read requests to prefetch.
+            When this is ``None`` (the default), do not limit the number of
+            concurrent prefetch requests.
```

Comment:
Should we add here that the openssl sftp client limits this to 64? Or maybe to the prefetch function docs only?

CC: @bskinn

Reply:
It does sound like it should be mentioned somewhere.

Is this limit a concern because users might run into difficulties in some situations if they choose a value higher than 64?

Or is this value of 64 merely a reference value, that's a recommended-not-to-exceed value?

Reply:
IMO it's a bit of both -- I think this issue is encountered when Paramiko *can't* transfer a file, but OpenSSL's sftp *can*, and that's because sftp is limiting concurrent requests (to 64). I don't know if there's anything particularly special/magical about the number 64, but since that's what sftp uses (and sftp generally works) it's worth mentioning as a reference.

Reply:
Ahh, if I understand your explanation, then definitely. A warning, something along the lines of "Note that OpenSSL `sftp` internally imposes a limit of 64 concurrent requests; setting a higher value here may cause problems." ?

Reply:
So, the current behavior (and default for this option) is "don't limit concurrent requests", which means that (if I'm doing my math right) requesting any file > 2MB (32kb chunks * 64 requests) already exceeds this limit (and works fine 99+% of the time). Because of that, I don't think "setting a higher value here may cause problems" is the right language. Possibly: "Note that OpenSSL `sftp` internally imposes a limit of 64 concurrent requests, while Paramiko imposes no limit by default; consider setting a limit if a file can be successfully received with `sftp` but experiences issues with Paramiko."

Reply:
Gotcha, sounds great!

Reply:
Do you want me to add that language inline here, or does that go somewhere else in the docs?

Reply:
Got it -- just added to `get` and `getfo` in sftp_client, as well as `prefetch` and `readv` in sftp_file as part of the parameter description.